### PR TITLE
Remove deprecated functions

### PR DESCRIPTION
**Pull request by nojb**
This PR proposes removing a bunch of deprecated functions in the 5.0 release.

### CODE REVIEW

**Comment by avsm** [comment]
For the `Obj.truncate` removal, I notice that there are a few other places where Obj.truncate has hindered some optimisations in the past:

```
asmcomp/cmm_helpers.ml
649:  (* We cannot deem this as [Immutable] due to the presence of [Obj.truncate]
middle_end/semantics_of_primitives.ml
77:      No_effects, Has_coeffects  (* That old chestnut: [Obj.truncate]. *)
```

It's probably better to fix these separately from this PR, but worth a followup issue to not forget?

**Review by damiendoligez — APPROVED** [review]
LGTM

**Comment by nojb** [comment]
> LGTM

Thanks, planning to merge by end of day unless someone objects.

**Comment by nojb** [comment]
> It's probably better to fix these separately from this PR, but worth a followup issue to not forget?

Thanks @avsm, will make a new issue with your note once the PR is merged.

**Review by avsm — APPROVED** [review]
I read through all these and it looks good to me as well.

**Comment by Octachron** [comment]
I have some doubts about the tag-related function in `Format`: the semantic tags were only introduced with 4.08, they are more useful than string tags in many cases but not all cases. However, the string tags are trivially implemented with the semantic tags, thus it might be better that I write a compatibility format library for potentially remaining users of string tags.

**Comment by nojb** [comment]
> I have some doubts about the tag-related function in `Format`: the semantic tags were only introduced with 4.08, they are more useful than string tags in many cases but not all cases. However, the string tags are trivially implemented with the semantic tags, thus it might be better that I write a compatibility format library for potentially remaining users of string tags.

Thanks for chiming in @Octachron! Sorry if I missed something, but are you suggesting any changes to the present PR, or was your comment just to float the idea of a compatibility library? I never use such libraries myself, but I don't have any objections to having one either :)

**Comment by Octachron** [comment]
Sorry, I meant to say that the deprecated string tag functions might have still have some active users in the wild: the deprecation is relatively recent (only 3 years) and the new API can be slightly more cumbersome in very specific cases. Thus I expect that this specific change will break some existing code somewhere.

However, since those deprecated functions can be easily implemented in a library, the simplest choice seems to offer a compatibility library for people that don't or can't move to the new semantic tags.

In short,  I am offering to write a compatibility library without changing anything in this PR (beyond keeping a record in this very discussion that those functions might require a compatibility library).

**Comment by gasche** [comment]
Note: some code breakage due to the removal of the Format string-tag API is being discussed in #10917.
